cpp/src/arrow/ipc/options.h

```diff
@@ -79,6 +80,10 @@ struct ARROW_EXPORT IpcReadOptions {
   /// deserializing RecordBatch. If null, return all deserialized fields
   util::optional<std::vector<int>> included_fields;
 
+  /// \brief Use global CPU thread pool to parallelize any computational tasks
+  /// like decompression
+  bool use_threads = true;
```

Comment:
Should the dataset IPC adapter be changed to disable this (because of nested parallelism)? @fsaintjacques

Reply:
Yes. Re threading discussion: Ideally `Read()` could yield the per-field decompression tasks so that `dataset::IpcFileFormat` could take advantage of parallelism even when scanning a single file